roachtest: add hibernate-spatial test

otan opened the pull request:
* Bump hibernate to 5.4.30
* Add hibernate-spatial test which tests cockroachdb against hibernate's
  spatial suite. Used a separate suite because the directory magic of
  copying may not work since the set of running tests overlap a bit.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/62492)
<!-- Reviewable:end -->

otan committed:
roachtest: add hibernate-spatial test

* Bump hibernate to 5.4.30
* Add hibernate-spatial test which tests cockroachdb against hibernate's
  spatial suite. Used a separate suite because the directory magic of
  copying may not work since the set of running tests overlap a bit.

Release note: None

rafiss reviewed (approved):
amazeballs

otan commented:
bors r=rafiss
